src/PhpDoc/TypeNodeResolver.php

```diff
@@ -823,7 +823,13 @@ private function resolveArrayShapeNode(ArrayShapeNode $typeNode, NameScope $name
 			$builder->setOffsetValueType($offsetType, $this->resolve($itemNode->valueType, $nameScope), $itemNode->optional);
 		}
 
-		return $builder->getArray();
+		$arrayType = $builder->getArray();
+		if ($typeNode->kind === ArrayShapeNode::KIND_LIST) {
+			$arrayType = AccessoryArrayListType::intersectWith($arrayType);
+			var_dump($arrayType);
```

Comment:
@ondrejmirtes `AccessoryArrayListType::intersectWith($arrayType)` never returns `IntersectionType` so changing `IntersectionType::describeItself()` doesn't seem to accomplish what you want, any ideas?

Reply:
`array{0: 'a', 1: 'b'}` is always a list so that's why the `&list` part is omitted and that's why it's not an intersection type. So we probably don't need to worry about this at all. The point of `list{...}` should be to reject types like `list{foo: 1}`.

Reply:
ok.